### PR TITLE
Add UpdateScale and GetScale methods

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

Addresses comment https://github.com/prometheus-operator/prometheus-operator/pull/5962#discussion_r1430123922

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by simonpasquier** [comment]
The test fails :thinking: 
```
prometheus_test.go:5082: failed to scale Prometheus allns-y-prometheusstatusscale-s6je32-0-9b070934/test: the name of the object (test based on URL) was undeterminable: name must be provided
````

**Comment by ArthurSens** [comment]
> The test fails 🤔
> 
> ```
> prometheus_test.go:5082: failed to scale Prometheus allns-y-prometheusstatusscale-s6je32-0-9b070934/test: the name of the object (test based on URL) was undeterminable: name must be provided
> ```

Interesting... the generated GetScale() method returns an error if .spec.shards is nil 🤔... not sure how to work around that

**Comment by simonpasquier** [comment]
Hmm, I guess we could have a default value of 1 for `.spec.shards` but it might break some workflows (e.g. reconcilers doing a raw diff between the desired spec and the current spec may trip over the API returning a shard value of 1 instead of null). What happens if the e2e tests set the shard value explicitly to 1?
We could add the default value for the PrometheusAgent CRD though since it's v1alpha1.

**Comment by ArthurSens** [comment]
> Hmm, I guess we could have a default value of 1 for `.spec.shards` but it might break some workflows (e.g. reconcilers doing a raw diff between the desired spec and the current spec may trip over the API returning a shard value of 1 instead of null). What happens if the e2e tests set the shard value explicitly to 1? 

They pass :)

> We could add the default value for the PrometheusAgent CRD though since it's v1alpha1.

That good, but since both CRDs use the same status/scale spec, we probably want to change both 😬

**Review by simonpasquier — APPROVED** [review]

File: test/e2e/prometheus_test.go
Comment (simonpasquier) [inline test/e2e/prometheus_test.go:1]: (nit) should we rename ` testPromScaleUpDownCluster` to `testPrometheusScaleUpDownReplicas` for clarity?

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**